backend/app/groups/service.py

```
@@ -193,7 +210,25 @@ async def get_group_by_id(self, group_id: str, user_id: str) -> Optional[dict]:
             logger.error(f"Unexpected error converting group_id to ObjectId: {e}")
             return None
 
-        group = await db.groups.find_one({"_id": obj_id, "members.userId": user_id})
+        # Convert user_id to ObjectId for querying both formats
+        try:
+            user_obj_id = ObjectId(user_id)
+        except:
```

Ruff (0.14.11)

Do not use bare `except`

(E722)